service-rpc/if/TCLIService.thrift

```diff
@@ -751,6 +751,52 @@ struct TGetTypeInfoResp {
   2: optional TOperationHandle operationHandle
 }
 
+// UploadData()
+//
+// UploadData data to table/path.
+struct TUploadDataReq {
+  // The session to execute the statement against
+  1: required TSessionHandle sessionHandle
+
+  // The table to be stored
+  2: optional string tableName
```

Comment:
Should we make it more explicit and say that one of `tableName` or `path` must be set?

Reply:
Added it:
https://github.com/apache/hive/pull/2878/files#diff-a97fa2c6efce03cd14924cbff678ef60b1846be6ff7f8275169a2f4d8799f16bR778